pyodide-build/pyodide_build/pypabuild.py

```diff
@@ -0,0 +1,108 @@
+import contextlib
+import os
+import sys
+import traceback
+from itertools import chain
+from pathlib import Path
+from typing import Mapping
+
+from build import BuildBackendException, ProjectBuilder  # type: ignore[import]
+from build.__main__ import (  # type: ignore[import]
+    _STYLES,
+    _error,
+    _handle_build_error,
+    _IsolatedEnvBuilder,
+    _ProjectBuilder,
+)
+from build.env import IsolatedEnv  # type: ignore[import]
+from packaging.requirements import Requirement
+
+from .common import get_hostsitepackages, get_pyversion
+
+UNISOLATED_PACKAGES = ["numpy", "scipy", "cffi", "pycparser", "pythran", "cython"]
+
+
+def symlink_unisolated_packages(env: IsolatedEnv):
+    pyversion = get_pyversion()
+    site_packages_path = f"lib/{pyversion}/site-packages"
+    env_site_packages = Path(env._path) / site_packages_path
```

Comment:
We do need access to `env._path` which is private to `IsolatedEnv`. Maybe we should request that `pypa/build` should make `path` public, or otherwise add functionality like this.

Reply:
There's a rewrite for IsolatedEnv coming that the ideal person to review hasn't had time for. I might review it in light of this (I've also become a little more familiar with this part of build recently, so I might be able to review it too).